src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs

```diff
@@ -565,6 +593,45 @@ public SyntaxAndDeclarationManager ReplaceSyntaxTree(SyntaxTree oldTree, SyntaxT
                 this.MessageProvider,
                 this.IsSubmission,
                 state);
+
+            OneOrMany<ImmutableSegmentedHashSet<string>> tryGetLastComputedMemberNames()
+            {
+                var previousRootNamespaceDeclaration = declMapBuilder[oldTree];
+                if (previousRootNamespaceDeclaration.IsValueCreated)
+                {
+                    // we computed the last root.  It will have up to date member names.  So just return those if present.
+                    var stack = s_declarationStack.Allocate();
+                    stack.Push(previousRootNamespaceDeclaration.Value);
+
+                    var builder = ArrayBuilder<ImmutableSegmentedHashSet<string>>.GetInstance();
+
+                    while (stack.Count > 0)
```

Comment:
>while (stack.Count > 0) [](http://example.com/codeflow?start=20&length=23)

Please convert to ```do {} while``` we know that the stack is not empty at the start.

Reply:
interesting idea.  done.